finality-aleph/src/network.rs

```diff
@@ -212,6 +227,23 @@ impl Peers {
     fn get(&self, session_id: SessionId, node_id: NodeIndex) -> Option<&PeerId> {
         self.to_peer.get(&session_id)?.get(&node_id)
     }
+
+    fn get_rand(
+        &self,
+        session_id: SessionId,
+        amount: usize,
+        offset: usize,
+    ) -> impl Iterator<Item = &PeerId> {
+        let skip = if let Some(n_peers) = self.to_peer.get(&session_id).map(|hm| hm.len()) {
```

Comment:
I might be confused, but is this really needed if you use `cycle` below? Or are you worried about the performance of `skip`?

Reply:
I worry about the performance of `cycle`. It clones the iterator, so I'm afraid that it wouldn't jump straight to index `skip` but actually make all the skipped clones.